packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -590,7 +612,7 @@ function parseIf(
         );
     }
 
-    if (!ast.isExpression(ifAttribute.value)) {
+    if (!ast.isExpression(ifAttribute.value, ctx.config.experimentalComplexExpressions)) {
```

Comment:
Could we avoid passing the flag to `isExpression` and let `isExpression` handle both cases where complex expression is enabled or not?

Reply:
I'll investigate to double-check this is necessary. I would love to remove the ~dozen changes to `isExpression` invocations.

Reply:
I was able to resolve this [here](https://github.com/salesforce/lwc/pull/3376/files#r1133044427).